src/list/state.rs

```diff
@@ -345,6 +347,39 @@ impl<'a> ListState<'a> {
         Ok(())
     }
 
+    pub fn select_if_matches_search_query(&mut self) {
+        let idx = self
+            .app_state
+            .exercises()
+            .iter()
+            .filter(|exercise| match self.filter() {
+                Filter::None => true,
+                Filter::Done => exercise.done,
+                Filter::Pending => !exercise.done,
+            })
+            .enumerate()
+            .find_map(|(i, s)| {
+                if s.name.contains(self.search_query.as_str()) {
+                    Some(i)
+                } else {
+                    None
+                }
+            });
+
+        match idx {
+            Some(i) => {
+                let exercise_ind = i;
+                self.scroll_state.set_selected(exercise_ind);
+                self.update_rows();
+            }
+            None => {
+                let msg = String::from("[NOT FOUND]") + &self.message.clone();
+                self.message.clear();
+                self.message.push_str(&msg);
```

Comment:
Maybe we can just push `" (not found)"` to the existing message.

Reply:
done!